apps/desktop/src/lib/baseBranch/baseBranchService.ts

```diff
@@ -67,6 +68,41 @@ export class BaseBranchService {
 		});
 		await this.fetchFromRemotes();
 	}
+
+	async push(withForce?: boolean) {
+		this.loading.set(true);
+		try {
+			await invoke<void>('push_base_branch', {
+				projectId: this.projectId,
+				withForce
+			});
+		} catch (err: any) {
+			if (err.code === Code.DefaultTargetNotFound) {
+				// Swallow this error since user should be taken to project setup page
+				return;
+			} else if (err.code === Code.ProjectsGitAuth) {
+				showError('Failed to authenticate', err);
+			} else {
+				showError('Failed to push', err);
+			}
+			console.error(err);
+		}
+		await this.fetchFromRemotes();
+	}
+
+	async realignBaseBranch(): Promise<string | undefined> {
+		this.loading.set(true);
+		try {
+			const stashedConflicting = await invoke<string[]>('realign_base_branch', {
+				projectId: this.projectId
+			});
+			await this.fetchFromRemotes();
+			return getStashedConflictingMessage(stashedConflicting);
+		} catch (err: any) {
+			showError('Failed to realign base branch', err);
+			console.error(err);
+		}
+	}
```

Comment:
Do we even call this from the frontend? Doing a quick text search I'm not seeing it.

Reply:
I think it's a large diff that wasn't rendered. At least for me